cli/lib/kontena_cli.rb

```diff
@@ -87,6 +109,20 @@ def self.cli_root(*joinables)
       File.join(Kontena.root, 'lib/kontena/cli', *joinables)
     end
   end
+
+  def self.logger
+    return @logger if @logger
+    require 'logger'
+    @logger = Logger.new($LOG_TARGET, 'weekly')
+    @logger.level = ENV["DEBUG"] ? Logger::DEBUG : Logger::INFO
+    @logger.progname = 'CLI'
+    @logger.formatter = proc do |severity, datetime, progname, msg|
+      timestamp = (1000 * (Time.now.to_f - ($PREVIOUS_LOG_TIME || $KONTENA_START_TIME))).to_i.to_s + "ms"
+      $PREVIOUS_LOG_TIME = Time.now.to_f
+      sprintf("%-6s", timestamp) + " #{progname}: #{msg}\n"
```

Comment:
Timestamps for profiling are good, but I might prefer the default absolute timestamp formatting with µs resolution over only having relative ms timestamps for everything... They have more information (was this log from the last run, or something I tested yesterday?), but you it's still pretty easy to eyeball the relative durations:

```
D, [2017-05-04T14:25:10.029109 #2661] DEBUG -- Rpc::NodeServicePodHandler: pod_cache miss
D, [2017-05-04T14:25:10.039161 #2661] DEBUG -- Rpc::NodeServicePodHandler: pod_cache miss
D, [2017-05-04T14:25:10.043157 #2661] DEBUG -- Rpc::NodeServicePodHandler: pod_cache miss
D, [2017-05-04T14:25:10.051214 #2661] DEBUG -- Rpc::NodeServicePodHandler: pod_cache miss
D, [2017-05-04T14:25:10.068203 #2661] DEBUG -- Rpc::NodeServicePodHandler: pod_cache miss
```